diff --git a/scripts/cmake/doctest.cmake b/scripts/cmake/doctest.cmake
index 3f8234c..3f0bc7d 100644
--- a/scripts/cmake/doctest.cmake
+++ b/scripts/cmake/doctest.cmake
@@ -109,7 +109,7 @@ function(doctest_discover_tests TARGET)
   cmake_parse_arguments(
     ""
     ""
-    "TEST_PREFIX;TEST_SUFFIX;WORKING_DIRECTORY;TEST_LIST;JUNIT_OUTPUT_DIR"
+    "TEST_PREFIX;TEST_SUFFIX;TEST_SUBCOMMAND;WORKING_DIRECTORY;TEST_LIST;JUNIT_OUTPUT_DIR"
     "TEST_SPEC;EXTRA_ARGS;PROPERTIES;ADD_LABELS"
     ${ARGN}
   )
@@ -146,6 +146,7 @@ function(doctest_discover_tests TARGET)
             -D "TEST_ADD_LABELS=${_ADD_LABELS}"
             -D "TEST_PREFIX=${_TEST_PREFIX}"
             -D "TEST_SUFFIX=${_TEST_SUFFIX}"
+            -D "TEST_SUBCOMMAND=${_TEST_SUBCOMMAND}"
             -D "TEST_LIST=${_TEST_LIST}"
             -D "TEST_JUNIT_OUTPUT_DIR=${_JUNIT_OUTPUT_DIR}"
             -D "CTEST_FILE=${ctest_tests_file}"
diff --git a/scripts/cmake/doctestAddTests.cmake b/scripts/cmake/doctestAddTests.cmake
index e559893..71cecc2 100644
--- a/scripts/cmake/doctestAddTests.cmake
+++ b/scripts/cmake/doctestAddTests.cmake
@@ -3,6 +3,7 @@
 
 set(prefix "${TEST_PREFIX}")
 set(suffix "${TEST_SUFFIX}")
+set(subcommand ${TEST_SUBCOMMAND})
 set(spec ${TEST_SPEC})
 set(extra_args ${TEST_EXTRA_ARGS})
 set(properties ${TEST_PROPERTIES})
@@ -36,7 +37,7 @@ if("${spec}" MATCHES .)
 endif()
 
 execute_process(
-  COMMAND ${TEST_EXECUTOR} "${TEST_EXECUTABLE}" ${spec} --list-test-cases
+  COMMAND ${TEST_EXECUTOR} "${TEST_EXECUTABLE}" ${subcommand} ${spec} --list-test-cases
   OUTPUT_VARIABLE output
   RESULT_VARIABLE result
   WORKING_DIRECTORY "${TEST_WORKING_DIR}"
@@ -61,7 +62,7 @@ foreach(line ${output})
   if(${add_labels})
     # get test suite that test belongs to
     execute_process(
-      COMMAND ${TEST_EXECUTOR} "${TEST_EXECUTABLE}" --test-case=${test} --list-test-suites
+      COMMAND ${TEST_EXECUTOR} "${TEST_EXECUTABLE}" ${subcommand} --test-case=${test} --list-test-suites
       OUTPUT_VARIABLE labeloutput
       RESULT_VARIABLE labelresult
       WORKING_DIRECTORY "${TEST_WORKING_DIR}"
@@ -97,6 +98,7 @@ foreach(line ${output})
     "${prefix}${test}${suffix}"
     ${TEST_EXECUTOR}
     "${TEST_EXECUTABLE}"
+    ${subcommand}
     "--test-case=${test_name}"
     "${TEST_JUNIT_OUTPUT_PARAM}"
     ${extra_args}
